src/views/smart_table.rs

```diff
@@ -0,0 +1,219 @@
+                    let w = (v_w - b_w) / cx.scale_factor(); // total width
+
+                    let delta_x = (x - b_w) - self.limiters[i];
+
+                    let prev_limiter = if i == 0 { 0.0 } else { self.limiters[i - 1] };
+                    let next_limiter =
+                        if i == self.limiters.len() - 1 { w } else { self.limiters[i + 1] };
+
+                    // Update new limiter position
+
+                    if delta_x.is_sign_positive() {
+                        self.limiters[i] += delta_x;
+
+                        // Min width
+                        if next_limiter - self.limiters[i] < CELL_MIN_SIZE_PX {
+                            self.limiters[i] = next_limiter - CELL_MIN_SIZE_PX;
+                        }
+                    } else {
+                        // Min width
+                        self.limiters[i] += delta_x;
+                        if self.limiters[i] - prev_limiter < CELL_MIN_SIZE_PX {
+                            self.limiters[i] = prev_limiter + CELL_MIN_SIZE_PX;
+                        }
+                    }
+
+                    // Set new Sizes
+
+                    if i == 0 {
+                        self.sizes[i] = self.limiters[i];
+                    } else {
+                        self.sizes[i] = self.limiters[i] - prev_limiter;
+                    }
+
+                    if i == self.limiters.len() - 1 {
+                        self.sizes[i + 1] = w - self.limiters[i]
+                    } else {
+                        self.sizes[i + 1] = next_limiter - self.limiters[i];
+                    }
+                }
+            }
+
+            WindowEvent::MouseUp(b) => {
+                if *b == MouseButton::Left {
+                    cx.emit(SmartTableEvent::StopDrag)
+                }
+            }
+            _ => {}
+        });
+    }
+}
+
+pub struct SmartTableRow {}
+
+impl SmartTableRow {
+    pub fn new<'a, L>(
+        cx: &'a mut Context,
+        data: &'a Vec<&'a str>,
+        sizes: L,
+        header: bool,
+    ) -> Handle<'a, Self>
+    where
+        L: Lens<Target = Vec<f32>>,
+    {
+        Self {}
+            .build(cx, |cx| {
+                for (i, d) in data.iter().enumerate() {
+                    if i != 0 {
+                        let element =
+                            Element::new(cx).class("smart-table-divisor").class("vertical");
+                        if header {
+                            element.class("accent");
+                            ResizeHandle::new(cx, i - 1, true)
+                                .toggle_class("active", SmartTable::dragging.map(|d| d.is_some()));
+                        }
+                    }
+
+                    HStack::new(cx, |cx| {
+                        Label::new(cx, *d).hoverable(false);
+                    })
+                    .hoverable(false)
+                    .class("smart-table-row-data-container")
+                    .width(sizes.clone().map(move |v| {
```

Comment:
You shouldn't need to clone the lens now that they are `Copy` 😄

Reply:
Right! I'll make sure to change that one ;)